docs/rfcs/2022-12-09-ephemeral-conversations.md

```diff
@@ -0,0 +1,208 @@
+# Ephemeral conversations with existing contacts
+
+Ephemeral conversation inside existing conversation with stricter security properties.
+
+- additional level of encryption for message bodies & text, chat items content & text (more?) inside chat db
+
+- ephemeral conversation key not persisted - only stored in-memory of current chat session, cleared on exiting to background
+
+- separate tables for chat items and messages to avoid gaps in ids? TBC db pages on deletion
+
+- don't persist chat items and messages at all? if yes - how to support multiple ephemeral conversations with different contacts in the same session - holding chat items in memory may become expensive. though only "not yet seen" items may have to be held in memory - after opening ephemeral conversation no longer keep them in memory; in this case closing ephemeral conversation screen (not fully exiting but keeping it in list of current ephemeral conversations) and re-opening also does not restore chat items, though connection and key are preserved
+
+- if multiple ephemeral conversations are allowed - how to know they have new messages - should there be notifications for them? only local or push notifications too? not a regular notification but indication in chat list?
```

Comment:
I'd send a regular push notification, show in the list a sum of both, some changed icon, something like "message in ephemeral chat" in the notification.